rust/agama-lib/src/users/store.rs

```diff
@@ -53,10 +58,7 @@ impl<'a> UsersStore<'a> {
             password: settings.password.clone().unwrap_or_default(),
             ..Default::default()
         };
-        let (success, issues) = self.users_client.set_first_user(&first_user).await?;
-        if !success {
-            return Err(ServiceError::WrongUser(issues));
-        }
+        self.users_client.set_first_user(&first_user).await?;
```

Comment:
why is suddenly possible to ignore failed case?

Reply:
The D-Bus client `set_first_user` returns a tuple `(success,issues)` in band, as `zbus::Result<(bool, Vec<String>)>`
https://github.com/openSUSE/agama/blob/b9a73b57fc71f6eb47b3f98154ec61a31f4d57d9/rust/agama-lib/src/users/client.rs#L84

But the HTTP client `set_first_user` returns a `Result<(), ...>` where all the errors are moved to the `Err` alternative, and the `WrongUser` is handled on that lower level:
https://github.com/openSUSE/agama/blob/3dca6b5ed6a30389b0023aea9a1ea031934a5859/rust/agama-lib/src/users/http_client.rs#L26

(and it is the `?` in this line that propagates the `Err`)